pulp_ansible/app/migrations/0036_update_repository_content.py

```diff
@@ -9,56 +9,77 @@ def create_repository_content_for_deprecation(apps, schema_editor):
     AnsibleCollectionDeprecated = apps.get_model('ansible', 'AnsibleCollectionDeprecated')
     RepositoryContent = apps.get_model('core', 'RepositoryContent')
     RepositoryVersionContentDetails = apps.get_model('core', 'RepositoryVersionContentDetails')
-    db_alias = schema_editor.connection.alias
     repo_content = []
     counts_list = []
     count = defaultdict(int)
 
-    for deprecation in AnsibleCollectionDeprecated.objects.using(db_alias).all():
-        repo_content.append(RepositoryContent(
-                content=deprecation,
-                repository_id=deprecation.repository_id,
-                version_added_id=deprecation.version_added_id,
-            ))
-        count[deprecation.version_added_id] += 1
+    # Create RepositoryContent with deduplicated deprecation items
+    for name, namespace in AnsibleCollectionDeprecated.objects.values_list("name", "namespace").distinct():
+        new_deprecation = AnsibleCollectionDeprecated.objects.create(pulp_type="collection_deprecation", name=name, namespace=namespace)
+        for deprecation in AnsibleCollectionDeprecated.objects.filter(name=name, namespace=namespace, repository_id__isnull=False):
+            repo_content.append(RepositoryContent(
+                    content=new_deprecation,
+                    repository_id=deprecation.repository_id,
+                    version_added_id=deprecation.version_added_id,
+                ))
+            count[deprecation.version_added_id] += 1
+            if len(repo_content) >= 1024:
+                RepositoryContent.objects.bulk_create(repo_content)
+                repo_content.clear()
+    RepositoryContent.objects.bulk_create(repo_content)
+    repo_content.clear()
+    AnsibleCollectionDeprecated.objects.filter(repository_id__isnull=False).delete()
 
+    # Update Repository content counts
     for repo_version_id, total in count.items():
         count_obj = RepositoryVersionContentDetails(
-            content_type=deprecation.pulp_type,
+            content_type="collection_deprecation",
```

Comment:
```suggestion
            content_type="ansible.collection_deprecation",
```

Reply:
Thank you so much i would not have found this in a week.